exp/controllers/helpers.go

```diff
@@ -222,3 +226,51 @@ func MachinePoolToInfrastructureMapFunc(gvk schema.GroupVersionKind) handler.ToR
 		}
 	}
 }
+
+// AzureClusterToAzureMachinePoolsFunc is a handler.ToRequestsFunc to be used to enqueue
+// requests for reconciliation of AzureMachinePools.
+func AzureClusterToAzureMachinePoolsFunc(kClient client.Client, log logr.Logger) handler.ToRequestsFunc {
+	return func(o handler.MapObject) []reconcile.Request {
+		ctx, cancel := context.WithTimeout(context.Background(), reconciler.DefaultMappingTimeout)
+		defer cancel()
+
+		c, ok := o.Object.(*infrav1.AzureCluster)
+		if !ok {
+			log.Error(errors.Errorf("expected a AzureCluster but got a %T", o.Object), "failed to get AzureCluster")
+			return nil
+		}
+		logWithValues := log.WithValues("AzureCluster", c.Name, "Namespace", c.Namespace)
+
+		cluster, err := util.GetOwnerCluster(ctx, kClient, c.ObjectMeta)
+		switch {
+		case apierrors.IsNotFound(err) || cluster == nil:
+			logWithValues.Info("owning cluster not found")
+			return nil
+		case err != nil:
+			logWithValues.Error(err, "failed to get owning cluster")
+			return nil
+		}
+
+		labels := map[string]string{clusterv1.ClusterLabelName: cluster.Name}
+		mpl := &clusterv1exp.MachinePoolList{}
+		if err := kClient.List(ctx, mpl, client.InNamespace(c.Namespace), client.MatchingLabels(labels)); err != nil {
```

Comment:
I think you could you directly list AzureMachinePools by cluster label?

Reply:
Not sure what you mean... FWIW I also copy/pasted that one directly from the controller but happy to improve it if you see a good improvement

Reply:
I think @alexeldeib is saying is that instead of traversing MachinePools -> AzureMachinePools through the infra ref, just list AzureMachinePools via:

```go
ampl := &invfrav1exp.AzureMachinePoolList{}
if err := kClient.List(ctx, mpl, client.InNamespace(c.Namespace), client.MatchingLabels(labels)); err != nil {
    ...
}
```

I think that should work, but I'm not 100% sure.

Reply:
oh I see. Let me try that.

Reply:
ok done, PTAL